pkg/driver/gid_allocator.go

```diff
@@ -87,13 +60,66 @@ func (g *GidAllocator) removeFsId(fsId string) {
 	delete(g.fsIdGidMap, fsId)
 }
 
-// Initializes a heap inclusive of min & max
-func initHeap(min, max int) *IntHeap {
-	h := make(IntHeap, max-min+1)
-	val := min
-	for i := range h {
-		h[i] = val
-		val += 1
+func (g *GidAllocator) getUsedGids(ctx context.Context, fsId string) (gids []int64, err error) {
+	gids = []int64{}
+	accessPoints, err := g.cloud.ListAccessPoints(ctx, fsId)
+	if err != nil {
+		err = fmt.Errorf("failed to list access points: %v", err)
+		return
 	}
-	return &h
+	if len(accessPoints) == 0 {
+		return gids, nil
+	}
+	for _, ap := range accessPoints {
+		// This should happen only in tests - skip nil pointers.
+		if ap == nil {
+			continue
+		}
+		if ap != nil && ap.PosixUser == nil {
+			err = fmt.Errorf("failed to discover used GID because PosixUser is nil for AccessPoint: %s", ap.AccessPointId)
+			return
+		}
+		gids = append(gids, ap.PosixUser.Gid)
+	}
+	klog.V(5).Infof("Discovered used GIDs: %+v for FS ID: %v", gids, fsId)
+	return
+}
+
+func getNextUnusedGid(usedGids []int64, gidMin, gidMax int) (nextGid int, err error) {
+	requestedRange := gidMax - gidMin
+	var availableGids []int
+
+	availableGids = make([]int, gidMax-gidMin+1)
+	if requestedRange > 120 {
+		klog.Warningf("Requested GID range (%v:%v) exceeds EFS Access Point limit (%v) per Filesystem. Driver will not allocate GIDs outside of this limit.", gidMin, gidMax, ACCESS_POINT_PER_FS_LIMIT)
+		availableGids = make([]int, ACCESS_POINT_PER_FS_LIMIT)
+	}
+
+	val := gidMax
+	for i := range availableGids {
```

Comment:
Can we remove the loop and modify line no:106 as below  to improve the performance?
`for gid := gidMax; gid >= gidMin; gid--`

Reply:
Indeed, changed.